### ISSUE TITLE
Loss oscillates wildly and doesn't decrease when training ltx2.3 with only 1 sample

### ISSUE THREAD

**Issue by segatecm** [LongCatFan1/longcat-tuner#62]
Hi everyone,I'm currently testing the training of ltx2.3.I’m using only 1 sample for training (single-sample overfitting test), but during training, the loss keeps oscillating violently and fails to decrease.Is this behavior normal? 
What could be the possible causes and how can I fix it?
Any advice or suggestions would be greatly appreciated!Thanks in advance!

my config.toml

dataset_config = "config/dataset.toml"

ltx2_mode = "video"                 
ltx2_first_frame_conditioning_p = 1

fp8_base = true                   
fp8_scaled = false                
gradient_checkpointing = true     
gradient_checkpointing_cpu_offload = false  
sdpa = false                       
flash_attn = true
use_pinned_memory_for_block_swap = false    

network_module = "networks.lora_ltx2"
network_dim = 32                    
network_alpha = 32                  
lora_target_preset = "v2v"          
#ic_lora_strategy = "v2v"
caption_dropout_rate = 0

optimizer_type = "AdamW8bit"        
learning_rate = 5e-6
max_train_epochs = 400
save_every_n_epochs = 20
save_last_n_epochs = 10              

timestep_sampling = "shifted_logit_normal"

output_dir = "./output"
output_name = "ltx2_lora-simple"
logging_dir = "./logs"

log_with = "tensorboard"
log_tracker_name = "ltx2_train"

loss report:
steps:   0%|                                 | 1/400 [00:00<00:00, 442.34it/s, avr_loss=0.142, loss_a=n/a, loss_v=0.142]/mnt/miniconda/envs/ltx2/lib/python3.10/site-packages/torch/distributed/c10d_logger.py:83: UserWarning: barrier(): using the device under current context. You can specify `device_id` in `init_process_group` to mute this warning.
  return func(*args, **kwargs)

epoch 2/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 2
steps:   0%|▏                               | 2/400 [00:20<1:08:49, 10.38s/it, avr_loss=0.153, loss_a=n/a, loss_v=0.153]
epoch 3/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 3
steps:   1%|▏                               | 3/400 [00:41<1:31:28, 13.83s/it, avr_loss=0.431, loss_a=n/a, loss_v=0.431]
epoch 4/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 4
steps:   1%|▎                               | 4/400 [01:02<1:42:59, 15.60s/it, avr_loss=0.127, loss_a=n/a, loss_v=0.127]
epoch 5/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 5
steps:   1%|▍                               | 5/400 [01:23<1:49:48, 16.68s/it, avr_loss=0.122, loss_a=n/a, loss_v=0.122]
epoch 6/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 6
steps:   2%|▍                               | 6/400 [01:44<1:54:48, 17.48s/it, avr_loss=0.123, loss_a=n/a, loss_v=0.123]
epoch 7/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 7
steps:   2%|▌                               | 7/400 [02:06<1:58:02, 18.02s/it, avr_loss=0.121, loss_a=n/a, loss_v=0.121]
epoch 8/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 8
steps:   2%|▋                               | 8/400 [02:27<2:00:28, 18.44s/it, avr_loss=0.207, loss_a=n/a, loss_v=0.207]
epoch 9/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 9
steps:   2%|▋                               | 9/400 [02:48<2:02:08, 18.74s/it, avr_loss=0.124, loss_a=n/a, loss_v=0.124]
epoch 10/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 10
steps:   2%|▊                              | 10/400 [03:10<2:03:37, 19.02s/it, avr_loss=0.147, loss_a=n/a, loss_v=0.147]
epoch 11/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 11
steps:   3%|▊                              | 11/400 [03:31<2:04:34, 19.21s/it, avr_loss=0.124, loss_a=n/a, loss_v=0.124]
epoch 12/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 12
steps:   3%|▉                              | 12/400 [03:52<2:05:24, 19.39s/it, avr_loss=0.141, loss_a=n/a, loss_v=0.141]
epoch 13/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 13
steps:   3%|█                              | 13/400 [04:13<2:06:00, 19.54s/it, avr_loss=0.122, loss_a=n/a, loss_v=0.122]
epoch 14/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 14
steps:   4%|█                              | 14/400 [04:35<2:06:27, 19.66s/it, avr_loss=0.124, loss_a=n/a, loss_v=0.124]
epoch 15/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 15
steps:   4%|█▏                             | 15/400 [04:56<2:06:51, 19.77s/it, avr_loss=0.163, loss_a=n/a, loss_v=0.163]
epoch 16/400
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 16
steps:   4%|█▏                             | 16/400 [05:17<2:07:07, 19.86s/it, avr_loss=0.642, loss_a=n/a, loss_v=0.642]
epoch 17/400

**Comment by AkaneTendo25** [comment]
A few things in your config seem questionable to me:

- `learning_rate = 5e-6` is very low. Even full fine-tune usually requires at least `~1e-5`. With only 1 sample and 400 epochs, you’re really only getting ~400 optimizer steps, so it may just be learning very slowly.
- `lora_target_preset = "v2v"` is a pretty broad target. For debugging, better start with a narrower preset first (`t1v`).
- I do not recommend to use `fp8_base` without `fp8_scaled`, it potentially may lead to precision-related instability (espeicaly with such small LR). 

Generally speaking, loss alone is not a reliable indicator of overfitting in diffusion training. In a single-sample overfit test, the loss can remain noisy or nearly flat even while the model is learning. I'd put more attention on fixed-seed preview generations (or maybe validation-loss trends if available).

**Comment by segatecm** [comment]
Thank you for your reply.
I made the following changes to the configuration and retrained once:

fp8_base = true 
fp8_scaled = true 
ltx2_first_frame_conditioning_p = 0
lora_target_preset = "t2v"
learning_rate = 1e-4

After that, I tested with this single sample and found that the generated video differs significantly from the original video. For example, the character's appearance and posture are completely different.

I want to confirm if my understanding is correct: If I train on only one sample, as training continues, the final output should become very close to the original video.

However, it seems that's not happening right now. I'm not sure whether the configuration is still incorrect, or if it's because the number of training steps is insufficient.
Below is the loss graph from my training.

<img width="343" height="297" alt="Image" src="https://github.com/user-attachments/assets/f1dfca8d-66fb-4075-9a65-db5455ad690c" />
